### ISSUE TITLE
Setfit no support evaluate each epoch or step and save model each epoch or step

### ISSUE THREAD

**Issue by batman-do** [huggingface/setfit#313]
Hi everyone, can u give me about evaluate each epoch and save checkpoint model ? thanks everyone

**Comment by tomaarsen** [comment]
Hello!

I'm afraid that at this point it is not possible to perform evaluation after every epoch or every X iterations, nor is it possible to get model checkpoints during training. This is not implemented yet due to how the model is trained in two separate subsequent phases, but at some point in the future we may be able to provide support for this functionality.

I'm aware that it is a well requested feature.

- Tom Aarsen

**Comment by tpalomo-dev** [comment]
Hello

Thanks for the project man, i do want to add a question related to this one though, i'm currently trying to fine tune a model using this package, and i can't find the way to see in what part of the training it is. I see examples where it shows this:
Applying column mapping to training dataset
***** Running training *****
  Num examples = 3640
  Num epochs = 1
  Total optimization steps = 228
  Total train batch size = 16
Epoch:   0%|          | 0/1 [00:00<?, ?it/s]
Iteration:   0%|          | 0/228 [00:00<?, ?it/s]

But i don't see it in my jupyter notebook. I'm currently using the '0.6.0' version. Is something i can do to see the training?

Thanks again

**Comment by tomaarsen** [comment]
@tpalomo-dev
You might have received a warning like:
```
TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html
  from .autonotebook import tqdm as notebook_tqdm
```
somewhere in your outputs. After following the instructions there, i.e. `pip install ipywidgets`, try to restart your Jupyter environment and afterwards it should start working. Let me know if you have any more issues with it.

- Tom Aarsen

**Comment by tpalomo-dev** [comment]
Hello man, thanks for the reply, i deleted the message because the problem was that the training wasn't even getting started, and that is why i couldn't see the progress. I have 130 classes and 49 examples for each for training, so i think that combined with the 20 num_iterations made the preprocess long (well, 5 minutes but since it doesn't show i didn't knew what was happening).

Thanks again for the project and the quick reply

**Comment by tomaarsen** [comment]
No worries! In the most recent version (0.6.0), there *should* be a progress bar showing the training pair generation progress, but that may have not worked for you.

**Comment by davidsbatista** [comment]
Hi @tomaarsen, are there any planned development to perform evaluation after every epoch or every X iterations, and model checkpoints during training?

**Comment by tomaarsen** [comment]
Yes and yes, both are already implemented in https://github.com/huggingface/setfit/pull/265, i.e. the setup for the v1.0.0 release of SetFit. 
Sadly, it still requires a decent bit of tests and especially documentation before it's ready, and I'm struggling with finding the spare time to get it out.
If I'm being 100% realistic, then I think it'll be released before the end of November.

**Comment by davidsbatista** [comment]
thanks for the quick reply! I will keep a look on #265

**Comment by tomaarsen** [comment]
This will be handled when #439 gets merged and released later this week. Keep an eye out!

**Comment by tomaarsen** [comment]
Closed via #439
